src/packagedcode/go.py

```diff
@@ -0,0 +1,256 @@
+        homepage_url=homepage_url,
+        download_url=download_url,
+        code_view_url=homepage_url,
+        dependencies=package_dependencies
+    )
+
+
+####################################################################
+# Regrex expressions to parse different types of dependency
+parse_module = re.compile(
+    r'^module\s'
+    r'(?P<module>[a-z].*)'
+).match
+
+parse_module_name = re.compile(
+    r'^module\s.*/'
+    r'(?P<name>[^\s]*)'
+).match
+
+parse_require = re.compile(
+    r'^require\s'
+    r'.*/(?P<name>[^\s]*)'
+    r'\s'
+    r'(?P<version>[a-z].*)'
+).match
+
+parse_exclude = re.compile(
+    r'^exclude\s'
+    r'.*/(?P<name>[^\s]*)'
+    r'\s'
+    r'(?P<version>[a-z].*)'
+).match
+
+parse_dep_link = re.compile(
+    r'^.*/(?P<name>[^\s]*)'
+    r'\s'
+    r'(?P<version>[a-z].*)'
+).match
+
+"""
+https://golang.org/ref/mod#go.mod-files
+
+module is in the form
+module github.com/alecthomas/participle
+>>> p = module_parse('module github.com/alecthomas/participle')
+>>> assert p.group('module') == ('github.com/alecthomas/participle')
+
+require or exclude can be in the form
+require github.com/davecgh/go-spew v1.1.1
+or
+exclude github.com/davecgh/go-spew v1.1.1
+or
+github.com/davecgh/go-spew v1.1.1
+
+>>> p = parse_require('require github.com/davecgh/go-spew v1.1.1')
+>>> assert p.group('name') == ('go-spew')
+>>> assert p.group('version') == ('v1.1.1')
+
+>>> p = parse_exclude('exclude github.com/davecgh/go-spew v1.1.1')
+>>> assert p.group('name') == ('go-spew')
+>>> assert p.group('version') == ('v1.1.1')
+
+>>> p = parse_dep_link('github.com/davecgh/go-spew v1.1.1')
+>>> assert p.group('name') == ('go-spew')
+>>> assert p.group('version') == ('v1.1.1')
+"""
+
+def preprocess(line):
+    """
+    Removes the comment portion and excess spaces.
+    """
+    if "//" in line:
+        line = line[:line.index('//')]
+    line = line.strip()
+    return line
+
+
+def parse_gomod(location):
+    """
+    Method to handle go.mod files.
+    """
+    contents = io.open(location, encoding='utf-8', closefd=True).readlines()
+    gomod_data = {}
+    require = []
+    exclude = []
+
+    for i, line in enumerate(contents):
+        line = preprocess(line)
+        parsed_module = parse_module(line)
+        if parsed_module:
+            gomod_data['module'] = parsed_module.group('module')
+
+        parsed_module_name = parse_module_name(line)
+        if parsed_module_name:
+            gomod_data['name'] = parsed_module_name.group('name')
+            
+        parsed_require = parse_require(line)
+        if parsed_require:
+            line_req = [parsed_require.group('name'), parsed_require.group('version')]
+            require.append(line_req)
+
+        parsed_exclude = parse_exclude(line)
+        if parsed_exclude:
+            line_req = [parsed_exclude.group('name'), parsed_exclude.group('version')]
+            exclude.append(line_req)
+
+        if "require(" in line or "require (" in line:
```

Comment:
What are you trying to handle here? what if there are two or twelve spaces?

Reply:
I have updated this in changes :)